homeassistant/components/bmw_connected_drive/device_tracker.py

```diff
@@ -1,51 +1,83 @@
 """Device tracker for BMW Connected Drive vehicles."""
 import logging
 
-from homeassistant.util import slugify
+from homeassistant.components.device_tracker import SOURCE_TYPE_GPS
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
 
-from . import DOMAIN as BMW_DOMAIN
+from . import DOMAIN as BMW_DOMAIN, BMWConnectedDriveBaseEntity
+from .const import CONF_ACCOUNT
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_scanner(hass, config, see, discovery_info=None):
-    """Set up the BMW tracker."""
-    accounts = hass.data[BMW_DOMAIN]
-    _LOGGER.debug("Found BMW accounts: %s", ", ".join([a.name for a in accounts]))
-    for account in accounts:
-        for vehicle in account.account.vehicles:
-            tracker = BMWDeviceTracker(see, vehicle)
-            account.add_update_listener(tracker.update)
-            tracker.update()
-    return True
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the BMW ConnectedDrive tracker from config entry."""
+    account = hass.data[BMW_DOMAIN][config_entry.entry_id][CONF_ACCOUNT]
+    entities = []
 
+    for vehicle in account.account.vehicles:
+        entities.append(BMWDeviceTracker(account, vehicle))
+        if not vehicle.state.is_vehicle_tracking_enabled:
+            _LOGGER.info(
+                "Tracking is (currently) disabled for vehicle %s (%s), defaulting to unknown.",
```

Comment:
We don't end logging messages with period.
```suggestion
                "Tracking is (currently) disabled for vehicle %s (%s), defaulting to unknown",
```